File: Module/PlayWays/src/main/java-room-rank/com/module/playways/room/room/comment/model/CommentAiModel.java
package com.module.playways.room.room.comment.model;

import android.text.SpannableStringBuilder;

import com.common.utils.SpanUtils;
import com.module.playways.room.prepare.model.PlayerInfoModel;

/**
 * Ai机器人消息
 */
public class CommentAiModel extends CommentModel {

    public CommentAiModel(PlayerInfoModel ai, String text) {
        setCommentType(CommentModel.TYPE_AI);
        setUserInfo(ai.getUserInfo());
        setAvatarColor(CommentModel.AVATAR_COLOR);
        SpannableStringBuilder stringBuilder = new SpanUtils()
                .append("AI裁判 ").setForegroundColor(CommentModel.RANK_NAME_COLOR)
                .append(text).setForegroundColor(CommentModel.RANK_TEXT_COLOR)
                .create();
        setStringBuilder(stringBuilder);
    }
}

File: Module/PlayWays/src/main/java-room-rank/com/module/playways/room/room/comment/model/CommentAiModel.kt
package com.module.playways.room.room.comment.model

import com.common.utils.SpanUtils
import com.module.playways.room.prepare.model.PlayerInfoModel

/**
 * Ai机器人消息
 */
class CommentAiModel(ai: PlayerInfoModel, text: String) : CommentModel() {

    init {
        commentType = TYPE_AI
        userInfo = ai.userInfo
        avatarColor = AVATAR_COLOR
        var nameBuilder = SpanUtils()
                .append("AI裁判 ").setForegroundColor(RANK_NAME_COLOR)
                .create()
        this.nameBuilder = nameBuilder

        var stringBuilder = SpanUtils()
                .append(text).setForegroundColor(RANK_TEXT_COLOR)
                .create()
        this.stringBuilder = stringBuilder
    }
}
